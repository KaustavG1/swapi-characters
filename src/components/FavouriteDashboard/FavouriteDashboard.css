.blank {
  background-color: inherit;
  width: 40px;
}
