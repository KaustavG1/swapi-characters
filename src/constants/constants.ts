export const baseUri = "https://swapi.info/api";
export const people = "people";
export const planets = "planets";
export const films = "films";
export const starships = "starships";
export const localStorageKey = "favs";
