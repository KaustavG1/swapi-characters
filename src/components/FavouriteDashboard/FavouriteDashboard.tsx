import { useNavigate } from "react-router";
import Button from "../common/Button/Button";
import EmptyRecords from "../common/EmptyRecords/EmptyRecords";
import ListContainer from "../ListContainer/ListContainer";
import { useEffect, useState } from "react";
import { localStorageKey } from "../../constants/constants";
import { getLocalValue } from "../../utils/localStorageHelper";
import "./FavouriteDashboard.css";

function FavouriteDashboard() {
  const [currentFavList, setCurrentFavList] = useState<any[]>([]);
  const navigate = useNavigate();

  const handleBackClick = () => {
    navigate(`/`);
  };

  useEffect(() => {
    const favs = getLocalValue(localStorageKey);
    setCurrentFavList(favs);
  }, []);

  return (
    <>
      <div className="top-nav">
        <Button
          text="Go Back"
          className="pagination-buttons back-button"
          onClick={handleBackClick}
        />
        <div className="section-header">Favourites</div>
        <div className="blank" />
      </div>
      {Array.isArray(currentFavList) && currentFavList?.length > 0 ? (
        <ListContainer data={currentFavList} className="fav-list" />
      ) : (
        <EmptyRecords />
      )}
    </>
  );
}

export default FavouriteDashboard;
