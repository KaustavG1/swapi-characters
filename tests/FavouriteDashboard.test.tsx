import axios from "axios";
import { it, expect, vi, MockedFunction } from "vitest";
import { render, screen, waitFor } from "@testing-library/react";
import { BrowserRouter } from "react-router";
import { mockCharacterData } from "./__mocks__/mockCharacterData";
import FavouriteDashboard from "../src/components/FavouriteDashboard/FavouriteDashboard";
import { mockLocalStorage } from "./__mocks__/mockLocalStorage";
import { localStorageKey } from "../src/constants/constants";
import { mockFavList } from "./__mocks__/mockFavList";

vi.mock("axios");
(axios.get as MockedFunction<typeof axios>).mockResolvedValue({
  data: mockCharacterData,
});

Object.defineProperty(window, "localStorage", {
  value: mockLocalStorage,
});

afterEach(vi.clearAllMocks);

describe("FavouriteDashboard component", () => {
  it("renders 'FavouriteDashboard' component empty state", async () => {
    const { container } = render(
      <BrowserRouter>
        <FavouriteDashboard />
      </BrowserRouter>
    );
    const favourites = screen.getByText(/Favourites/i);
    const noRecordsMessage = screen.getByText(/No Records found./i);

    expect(favourites).toBeInTheDocument();
    expect(noRecordsMessage).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("renders 'FavouriteDashboard' component with record", async () => {
    mockLocalStorage.setItem(localStorageKey, JSON.stringify(mockFavList));
    render(
      <BrowserRouter>
        <FavouriteDashboard />
      </BrowserRouter>
    );

    await waitFor(() => {
      expect(screen.getAllByText(/Luke Skywalker/i)[0]).toBeInTheDocument();
    });
  });
});
