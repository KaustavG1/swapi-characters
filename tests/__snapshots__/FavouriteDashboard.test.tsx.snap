// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CharacterView component > renders 'CharacterView' component failure case 1`] = `
<div>
  <div
    class="blank-page"
  >
    <div
      class="loader"
    >
      Loading ...
    </div>
  </div>
</div>
`;

exports[`CharacterView component > renders 'CharacterView' component success case 1`] = `
<div>
  <div
    class="blank-page"
  >
    <div
      class="loader"
    >
      Loading ...
    </div>
  </div>
</div>
`;

exports[`FavouriteDashboard component > renders 'FavouriteDashboard' component 1`] = `
<div>
  <div
    class="top-nav"
  >
    <div>
      <button
        class="pagination-buttons back-button button-default"
      >
        <span />
        <span>
          Go Back
        </span>
      </button>
    </div>
    <div
      class="section-header"
    >
      Favourites
    </div>
    <div
      class="blank"
    />
  </div>
  <div
    class="blank-page"
  >
    <div
      class="empty-record-message"
    >
      No Records found.
    </div>
  </div>
</div>
`;

exports[`FavouriteDashboard component > renders 'FavouriteDashboard' component empty state 1`] = `
<div>
  <div
    class="top-nav"
  >
    <div>
      <button
        class="pagination-buttons back-button button-default"
      >
        <span />
        <span>
          Go Back
        </span>
      </button>
    </div>
    <div
      class="section-header"
    >
      Favourites
    </div>
    <div
      class="blank"
    />
  </div>
  <div
    class="blank-page"
  >
    <div
      class="empty-record-message"
    >
      No Records found.
    </div>
  </div>
</div>
`;

exports[`FavouriteDashboard component > renders 'FavouriteDashboard' component success case 1`] = `
<div>
  <div
    class="top-nav"
  >
    <div>
      <button
        class="pagination-buttons back-button button-default"
      >
        <span />
        <span>
          Go Back
        </span>
      </button>
    </div>
    <div
      class="section-header"
    >
      Favourites
    </div>
    <div
      class="blank"
    />
  </div>
  <div
    class="blank-page"
  >
    <div
      class="empty-record-message"
    >
      No Records found.
    </div>
  </div>
</div>
`;
